### PR TITLE
Refactored `ThemeColors` into an asset

### PR DESCRIPTION
**Pull request by Fincap**
- Annotated `ThemeColors` with appropriate serde derive macros to allow serialization and deserialization to/from a `material-theme.json` generated by the [Material Theme Builder](https://material-foundation.github.io/material-theme-builder/).
- `ThemeColors` now derives `Asset`, and has a simple asset loader which uses `serde_json` to generate the struct.
- Created an example `custom_theme` to demonstrate loading and using a custom theme from `material-theme.json`.

To get this to work without writing an overly-complicated custom deserializer, I had to "flatten" the structure of `ColorScheme` into `ColorSchemes`. This only impacted the `ThemeData::contrast` method, but might also impact any users of the create who reference `ColorScheme` directly (of which I doubt there are many).

Happy to take on any feedback for the changes :)

### CODE REVIEW

**Comment by eidloi** [comment]
I can't begin to describe how happy I am that you made this, thank you! I'll do a review this evening, though it looks alright at a glance, cheers!

File: crates/sickle_ui_scaffold/src/theme/theme_colors.rs [hunk 1]
```diff
@@ -262,10 +427,12 @@ pub struct ColorPalettes {
 
 // TODO: write asset loader for theme colors and load it from a material-theme.json
```
Comment (eidloi) [inline crates/sickle_ui_scaffold/src/theme/theme_colors.rs:428]: This TODO can go now! This PR also makes the theme colors follow the format (not just loosly anymore)

File: crates/sickle_ui_scaffold/src/theme/theme_colors.rs [hunk 2]
```diff
@@ -696,3 +861,82 @@ impl Default for ThemeColors {
         }
     }
 }
+
+/// Custom serialization and deserialization functions necessary for the loading and saving of
+/// [`Color`] structs to their hex string representation.
+mod serialize_color {
```
Comment (eidloi) [inline crates/sickle_ui_scaffold/src/theme/theme_colors.rs:867]: I am perplexed to have these so deep in the file (after that chonky default). The mods better go up top after imports or separate files (in this case they can go up top).

File: crates/sickle_ui_scaffold/src/theme/theme_data.rs [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
-use crate::ui_style::builder::StyleBuilder;
-
 use super::{
     icons::Icons,
     style_animation::AnimationSettings,
     theme_colors::{SchemeColors, ThemeColors},
     theme_spacing::ThemeSpacing,
     typography::ThemeTypography,
 };
+use crate::ui_style::builder::StyleBuilder;
```
Comment (eidloi) [inline crates/sickle_ui_scaffold/src/theme/theme_data.rs:11]: this is just me, but I prefer the imports to be from "further away" to "more closer" to the file. Meaning: std, external crates, internal crates, super / sibling, own mods (, and inline mods after). Hence the original order. If it was formatted automatically you can leave it like this (I _prefer_ most of all all the automated formatters with default configs like a lazy bum :D)

**Review by eidloi — COMMENTED** [review]
Just some nit picking, otherwise it is wonderful!

File: crates/sickle_ui_scaffold/src/theme/theme_data.rs (hunk 3 above)
Comment (Fincap) [reply crates/sickle_ui_scaffold/src/theme/theme_data.rs:11]: Whoops, I forgot to suppress my global rustfmt config 😅 I use `group_imports = "StdExternalCrate"` which is an experimental/nightly config. I'll move it back 😛

It looks like I also accidentally wrapped the comments in this file to 100 characters. I'll leave it as-is for now (out of selfishness 😉), but I'll revert those too if you prefer.

Comment (eidloi) [reply crates/sickle_ui_scaffold/src/theme/theme_data.rs]: I can't wait for such formatting options to make their way into stable! No need to fix the comments, I manually break them as I see fit so they are all over the place.

**Review by eidloi — APPROVED** [review]

**Comment by eidloi** [comment]
Excellent work! Thank you!

**Comment by Fincap** [comment]
No worries! Happy to help, this is a great crate with a lot of potential 😃

**Pull request merged**